qevy-derive/src/entities.rs

```diff
@@ -0,0 +1,233 @@
+        .iter()
+        .map(|(_, base_class_name)| base_class_name);
+
+    // define impl variables
+    let ident = &ast.ident;
+    // if the entity_name is not set, use the struct name
+    let struct_name = entity_name.unwrap_or_else(|| ident.to_string());
+    let (impl_generics, type_generics, where_clause) = ast.generics.split_for_impl();
+
+    // Generate code
+    let generated_code = quote::quote!(
+        impl #impl_generics QevyEntity for #ident #type_generics #where_clause {
+            fn get_export_string(
+                &self,
+                my_registration: &bevy::reflect::TypeRegistration,
+                registry: &bevy::reflect::TypeRegistry,
+                default_value: &mut Box<dyn Reflect>,
+            ) -> String {
+                let type_info = my_registration.type_info();
+                let short_name = #struct_name;
+
+                let field_names: Vec<&str> = vec![#(#field_names),*];
+                let field_comments: Vec<&str> = vec![#(#field_comments),*];
+
+                let size_string = #entity_size_string;
+                let color_string = #color_string;
+                let model_string: &str = #model_string;
+
+                let base_classes_field_names: Vec<&str> = vec![#(#base_classes_field_names),*];
+                let base_classes_base_class_names: Vec<&str> = vec![#(#base_classes_base_class_names),*];
+                let joined_base_classes = base_classes_base_class_names.join(", ");
+                let base_class_string = if joined_base_classes.is_empty() {
+                    String::new()
+                } else {
+                    format!("base({})", joined_base_classes)
+                };
+                let description = #entity_description;
+                let entity_type = #entity_type;
+
+                let types_string = match type_info {
+                    bevy::reflect::TypeInfo::Struct(info) => {
+                        let mut types_string = String::new();
+
+                        for named_field in info.iter() {
+                            let name = named_field.name();
+
+                            // Ignore base classes, as they don't need their own field in the fgd
+                            if base_classes_field_names.iter().any(|&base_class| base_class == name) {
+                                continue;
+                            }
+
+                            let index_of_field_name = field_names.iter().position(|s| s == &name).expect(format!("Field name not found: {}", name).as_str());
+                            let description = field_comments[index_of_field_name];
+
+                            let field_type_id = named_field.type_id();
+                            let field_registry = registry.get(field_type_id).expect(format!("Field type not found: {}", name).as_str());
+
+                            let ReflectMut::Struct(mut_value) = default_value.reflect_mut() else {
```

Comment:
```suggestion
                            let bevy::reflect::ReflectMut::Struct(mut_value) = default_value.reflect_mut() else {
```

If we fully qualify here so we don't need to import it when using `#[derive(QevyEntity)]`